narwhal/consensus/src/tests/bullshark_tests.rs

```diff
@@ -818,3 +822,257 @@ async fn restart_with_new_committee() {
+    // to previous rounds between them are fully connected - meaning that we always add all the parents
+    // from previous round, except for the authority 4 which we consider it to be slow, so no one
+    // refers to their certificates. Authority 4 still produces certificates, and uses as parents
+    // all the certificates of the other authorities but never anyone else is referring to its
+    // certificates. That will create a lone chain for authority 4. We should not see any certificate
+    // committed for authority 4.
+    let keys: Vec<PublicKey> = committee
+        .authorities()
+        .map(|(key, _)| key.clone())
+        .collect();
+    let slow_node = keys[3].clone();
+    let genesis = Certificate::genesis(&committee);
+
+    let slow_nodes = vec![(slow_node.clone(), 0.0_f64)];
+    let (certificates, round_8_certificates) = test_utils::make_certificates_with_slow_nodes(
+        &committee,
+        1..=8,
+        genesis,
+        &keys,
+        slow_nodes.as_slice(),
+    );
+
+    let mut certificates: VecDeque<Certificate> = certificates;
+    let mut slow_node_certificates = VecDeque::new();
+
+    // Now we keep only the certificates from authorities 1-3
+    certificates.retain(|c| {
+        if c.origin() == slow_node {
+            // if it is slow node's add it to the dedicated vec
+            slow_node_certificates.push_back(c.clone());
+            return false;
+        }
+        true
+    });
+
+    // Create Bullshark consensus engine
+    let store = make_consensus_store(&test_utils::temp_dir());
+    let metrics = Arc::new(ConsensusMetrics::new(&Registry::new()));
+    let mut state = ConsensusState::new(metrics.clone());
+    let mut bullshark = Bullshark::new(
+        committee.clone(),
+        store,
+        GC_DEPTH,
+        metrics,
+        NUM_SUB_DAGS_PER_SCHEDULE,
+    );
+
+    // Now start feeding the certificates per round up to 8. We expect to have
+    // triggered a commit up to round 6 and gc round 1.
+    for c in certificates {
+        let _ = bullshark
+            .process_certificate(&mut state, c.clone())
+            .unwrap();
+    }
+
+    // We expect everything to have been cleaned up by standard gc until round 3 (included)
+    assert_eq!(
+        state
+            .dag
+            .iter()
+            .filter(|(round, _)| **round <= 1_u64)
+            .count(),
+        0,
+        "Didn't expect to still have certificates from round 1 and 2"
+    );
+
+    // Now send the certificates of slow node. Now leader election can happen for round 8
+    // as we haven't sent yet certificates of round 9
+    for c in slow_node_certificates {
+        let _ = bullshark.process_certificate(&mut state, c).unwrap();
+    }
+
+    // Now create the certificates of round 9, and ensure everyone gives support to
+    // leader of round 8 (the slow node - 4) so we can trigger a commit. Also since slow node
```

Comment:
Can you just add a note how we are determining leader of round 8 as 4 deterministically due to your round robin leader changes for tests. For future test readers who don't have the PR context